### PR TITLE
Added TensorFlow project with 11 fuzz targets.

### PR DESCRIPTION
**Pull request by kobrineli**

### CODE REVIEW

File: projects/tensorflow/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Google Inc.
+#
```
Comment (SweetVishnya) [inline projects/tensorflow/Dockerfile:2]: Modification copyright here and in other files

File: projects/tensorflow/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ENV CFLAGS "-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
+ENV CXXFLAGS_EXTRA "-stdlib=libc++"
+ENV CXXFLAGS "$CFLAGS $CXXFLAGS_EXTRA"
+
+ENV JAVA_HOME /usr/lib/jvm/java-15-openjdk-amd64
+# it fails after installing java due to jazzer env variables missing
+RUN install_java.sh || true
+
+WORKDIR /
+RUN git clone --depth 1 https://github.com/tensorflow/tensorflow tensorflow
```
Comment (SweetVishnya) [inline projects/tensorflow/Dockerfile:63]: Checkout specific commit here and for Sydr

File: projects/tensorflow/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Run Hybrid Fuzzing
+
+Unzip Sydr (`sydr.zip`) in `projects/tensorflow` directory:
+
+	$ unzip sydr.zip
+
+Run Docker:
+
+	$ docker run --rm -it -v `pwd`:/fuzz oss-sydr-fuzz-tensorflow /bin/bash
```
Comment (SweetVishnya) [inline projects/tensorflow/README.md:17]: wrong docker command, check other projects

File: projects/tensorflow/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+Run hybrid fuzzing
+
+	# sydr-fuzz -c cleanpath_fuzz.toml run -l debug
+
+## Alternative Fuzz Targets
+
+TensorFlow project has 11 fuzz targets.
+
+### arg\_def\_case\_fuzz
```
Comment (SweetVishnya) [inline projects/tensorflow/README.md:31]: Backslashes are not necessary

File: projects/tensorflow/arg_def_case_fuzz.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+[sydr]
+args = "-l debug --optimistic --invert-n 1024 --app-timeout 60"
```
Comment (SweetVishnya) [inline projects/tensorflow/arg_def_case_fuzz.toml:18]: --optimistic is not necessary, it is auto inserted

File: projects/tensorflow/base64_fuzz.toml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+ Copyright 2021 ISP RAS
```
Comment (SweetVishnya) [inline projects/tensorflow/base64_fuzz.toml:1]: wrong comment

File: projects/tensorflow/build_fuzz.sh [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    then
+    echo "--linkopt=$(find $(llvm-config --libdir) -name libclang_rt.ubsan_standalone_cxx-x86_64.a | head -1)"
+    fi
+done
+)"
+
+# Ugly hack to get LIB_FUZZING_ENGINE only for fuzz targets
+# and not for other binaries such as protoc
+sed -i -e 's/linkstatic/linkopts = \["-fsanitize=fuzzer,address,undefined,integer,null,float-divide-by-zero,bounds"\],\nlinkstatic/' tensorflow/security/fuzzing/tf_fuzzing.bzl
+#sed -i -e 's/linkstatic/linkopts = \["-fsanitize=fuzzer"\],\nlinkstatic/' tensorflow/security/fuzzing/tf_fuzzing.bzl
```
Comment (SweetVishnya) [inline projects/tensorflow/build_fuzz.sh:63]: redundant comment

File: projects/tensorflow/build_fuzz.sh [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  # symlink that ends up pointing to itself!
+  pushd bazel-tensorflow
+  [[ -e external/org_tensorflow ]] && unlink external/org_tensorflow
+  ${RSYNC_CMD} external/ ${REMAP_PATH}
+  popd
+fi
+
+# Finally, make sure we don't accidentally run with stuff from the bazel cache.
+rm -f bazel-*
+#rm -rf ~/.cache
```
Comment (SweetVishnya) [inline projects/tensorflow/build_fuzz.sh:119]: uncomment or remove

File: projects/tensorflow/build_fuzz.sh [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+#bazel clean --expunge
+#git clean -dfx
```
Comment (SweetVishnya) [inline projects/tensorflow/build_fuzz.sh:122]: uncomment or remove

File: projects/tensorflow/cleanpath_fuzz.toml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+[sydr]
+args = "-l debug --invert-n 1024 --app-timeout 60"
+target = "/sydr/cleanpath_fuzz @@"
+jobs = 16
```
Comment (SweetVishnya) [inline projects/tensorflow/cleanpath_fuzz.toml:20]: Too many jobs for user pc

File: projects/tensorflow/string_replace_fuzz.toml [hunk 11]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (SweetVishnya) [inline projects/tensorflow/string_replace_fuzz.toml:1]: remove this line

File: projects/tensorflow/tstring_fuzz.toml [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+[sydr]
+args = "-l debug --invert-n 1024 --app-timeout 60"
+target = "/sydr/tstring_fuzz @@"
+
+[libfuzzer]
+path = "/fuzzer/tstring_fuzz"
+args = "-rss_limit_mb=8192 /fuzz/corpus"
+
+
```
Comment (SweetVishnya) [inline projects/tensorflow/tstring_fuzz.toml:25]: remove these lines

**Comment by SweetVishnya** [comment]
Also, add tensorflow to README.md

**Comment by SweetVishnya** [comment]
Screenshot from 2021-12-13 14-00-08

File: projects/tensorflow/tstring_fuzz.toml (hunk 12 above)
Comment (kobrineli) [reply projects/tensorflow/tstring_fuzz.toml:25]: Done

File: projects/tensorflow/string_replace_fuzz.toml (hunk 11 above)
Comment (kobrineli) [reply projects/tensorflow/string_replace_fuzz.toml:1]: Done

File: projects/tensorflow/cleanpath_fuzz.toml (hunk 10 above)
Comment (kobrineli) [reply projects/tensorflow/cleanpath_fuzz.toml:20]: Done

File: projects/tensorflow/build_fuzz.sh (hunk 9 above)
Comment (kobrineli) [reply projects/tensorflow/build_fuzz.sh:122]: Done

File: projects/tensorflow/build_fuzz.sh (hunk 8 above)
Comment (kobrineli) [reply projects/tensorflow/build_fuzz.sh:119]: Done

File: projects/tensorflow/build_fuzz.sh (hunk 7 above)
Comment (kobrineli) [reply projects/tensorflow/build_fuzz.sh:63]: Done

File: projects/tensorflow/arg_def_case_fuzz.toml (hunk 5 above)
Comment (kobrineli) [reply projects/tensorflow/arg_def_case_fuzz.toml:18]: Done

File: projects/tensorflow/base64_fuzz.toml (hunk 6 above)
Comment (kobrineli) [reply projects/tensorflow/base64_fuzz.toml:1]: Done

File: projects/tensorflow/README.md (hunk 4 above)
Comment (kobrineli) [reply projects/tensorflow/README.md:31]: Done

File: projects/tensorflow/README.md (hunk 3 above)
Comment (kobrineli) [reply projects/tensorflow/README.md:17]: Done

File: projects/tensorflow/Dockerfile (hunk 2 above)
Comment (kobrineli) [reply projects/tensorflow/Dockerfile:63]: Done

File: projects/tensorflow/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+WORKDIR /tensorflow
+RUN git reset --hard b56e6db5e7cccfffa824bbc1b5e018c6cc413c21
```
Comment (SweetVishnya) [inline projects/tensorflow/Dockerfile:66]: git checkout, для этого depth надо убрать у clone

File: projects/tensorflow/README.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+## Run Hybrid Fuzzing
+
+Unzip Sydr (`sydr.zip`) in `projects/tensorflow` directory:
+
+	$ unzip sydr.zip
+
+Run Docker:
+
+        $ sudo docker run -v /etc/localtime:/etc/localtime:ro --privileged --network host --rm -it -v $PWD:/fuzz oss-sydr-fuzz-tensorflow /bin/bash
```
Comment (SweetVishnya) [inline projects/tensorflow/README.md:17]: wrong indentation

File: projects/tensorflow/cleanpath_fuzz.toml [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+[sydr]
+args = "-l debug --invert-n 1024 --app-timeout 60"
+target = "/sydr/cleanpath_fuzz @@"
+jobs = 4
+
+[libfuzzer]
+path = "/fuzzer/cleanpath_fuzz"
+args = "-rss_limit_mb=8192 /fuzz/corpus"
+jobs = 4
```
Comment (SweetVishnya) [inline projects/tensorflow/cleanpath_fuzz.toml:25]: -jobs=4 in args

File: projects/tensorflow/build_sydr.sh [hunk 16]
```diff
@@ -0,0 +1,101 @@
+# Sydr
```
Comment (SweetVishnya) [inline projects/tensorflow/build_sydr.sh:1]: copyright with modifications

File: projects/tensorflow/Dockerfile (hunk 13 above)
Comment (kobrineli) [reply projects/tensorflow/Dockerfile]: Done

File: projects/tensorflow/README.md (hunk 14 above)
Comment (kobrineli) [reply projects/tensorflow/README.md]: Done

File: projects/tensorflow/cleanpath_fuzz.toml (hunk 15 above)
Comment (kobrineli) [reply projects/tensorflow/cleanpath_fuzz.toml]: Done

File: projects/tensorflow/build_sydr.sh (hunk 16 above)
Comment (kobrineli) [reply projects/tensorflow/build_sydr.sh:19]: Done

**Pull request merged**